app/models/filters/filter_by_repeatedness.rb

```diff
@@ -6,18 +6,18 @@ class FilterByRepeatedness < Filter
     validates :repeated_classifications, inclusion: {in: REPEATED_CLASSIFICATIONS}
 
     def apply(extract_groups)
+      ordered_extract_groups = extract_groups.sort_by(&:classification_at)
```

Comment:
Not blocking. 

As the `extract_group` param here is a result of the `ExtractsForClassification.from(extracts)` outlined in https://github.com/zooniverse/caesar/blob/68b50a8c4eec11981de6f433e1cb0a559fe6981c/app/models/extract_filter.rb#L13-L14 we could have added the sorting to that class behaviour perhaps where it's best placed to ensure the `ExtractGroups` know how to sort themselves...then this code is just responsible for the filtering based on the order as passed in. 

i.e. the `ExtractsForClassification.from` class method becomes
```
class ExtractsForClassification
  def self.from(extracts)
    extracts
      .sort_by(&:classification_at)
      .group_by(&:classification_id)
      .map { |_, group| new(group) }
  end
```
and as the specs rely on this for ordering and so does the actual data (as per the code flow i outlined above) then you end up with the same result..perhaps just a slight cleaner change but also ensuring the responsibilites of the classes are clear and not co-mingling, i.e. `ExtractsForClassification` prepares the extracts groups for filtering and `FilterByRepeatedness` does the repeat filtering.

Reply:
OH thats interesting! Yeah, it might have been buggy from the start since `filter_by_repeatedness` went first and then `filter_by_subrange` with the `sort_by` came after that. That's wild.